### ISSUE TITLE
Undefined method "should_be_stored"

### ISSUE THREAD

**Issue by CharlyCast** [hsahovic/poke-env#3]
### Expected behaviour:

When copy-pasting the "max-damage-player" into player.py, running `python player.py` should return something like 

```sh
Max damage player won 90 / 100 battles [this took 12.470762 seconds]
```

### Actual behaviour

The following error is thrown:

```sh
Task exception was never retrieved
future: <Task finished coro=<PlayerNetwork._handle_message() done, defined at /home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player_network_interface.py:111> exception=Attri
buteError("type object 'Move' has no attribute 'should_be_stored'")>
Traceback (most recent call last):
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player_network_interface.py", line 139, in _handle_message
    await self._handle_battle_message(message)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player.py", line 210, in _handle_battle_message
    await battle._parse_message(split_message)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/environment/battle.py", line 282, in _parse_message
    self.get_pokemon(pokemon)._moved(move)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/environment/pokemon.py", line 143, in _moved
    if Move.should_be_stored(id_):
AttributeError: type object 'Move' has no attribute 'should_be_stored'
Task exception was never retrieved
future: <Task finished coro=<PlayerNetwork._handle_message() done, defined at /home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player_network_interface.py:111> exception=Attri
buteError("type object 'Move' has no attribute 'should_be_stored'")>
Traceback (most recent call last):
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player_network_interface.py", line 139, in _handle_message
    await self._handle_battle_message(message)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player.py", line 210, in _handle_battle_message
    await battle._parse_message(split_message)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/environment/battle.py", line 282, in _parse_message
    self.get_pokemon(pokemon)._moved(move)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/environment/pokemon.py", line 143, in _moved
    if Move.should_be_stored(id_):
AttributeError: type object 'Move' has no attribute 'should_be_stored'
Task exception was never retrieved
future: <Task finished coro=<PlayerNetwork._handle_message() done, defined at /home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player_network_interface.py:111> exception=AttributeError("type object 'Move' has no attribute 'should_be_stored'")>
Traceback (most recent call last):
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player_network_interface.py", line 139, in _handle_message
    await self._handle_battle_message(message)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player.py", line 210, in _handle_battle_message
    await battle._parse_message(split_message)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/environment/battle.py", line 282, in _parse_message
    self.get_pokemon(pokemon)._moved(move)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/environment/pokemon.py", line 143, in _moved
    if Move.should_be_stored(id_):
AttributeError: type object 'Move' has no attribute 'should_be_stored'
Task exception was never retrieved
future: <Task finished coro=<PlayerNetwork._handle_message() done, defined at /home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player_network_interface.py:111> exception=AttributeError("type object 'Move' has no attribute 'should_be_stored'")>
Traceback (most recent call last):
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player_network_interface.py", line 139, in _handle_message
    await self._handle_battle_message(message)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/player/player.py", line 210, in _handle_battle_message
    await battle._parse_message(split_message)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/environment/battle.py", line 282, in _parse_message
    self.get_pokemon(pokemon)._moved(move)
  File "/home/gripsou/.local/lib/python3.7/site-packages/poke_env/environment/pokemon.py", line 143, in _moved
    if Move.should_be_stored(id_):
AttributeError: type object 'Move' has no attribute 'should_be_stored' 
```

**Comment by hsahovic** [comment]
Thank you for bringing that up.
I'm looking into it: it should be fixed soon.

**Comment by hsahovic** [comment]
This is fixed by #5, and tested. I also added integration tests.

**Comment by hsahovic** [comment]
This is fixed by #5; the PR also adds a test related to the issue.
